pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+              * `host`: Host and optionally port of the WBEM server containing
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+            * For class-level requests: A list of :class:`~pywbem.CIMClassName`
+              objects that are the class paths of the referencing association
+              classes, with their attributes set as follows:
+
+              * `classname`: Name of the class.
+              * `namespace`: Name of the CIM namespace containing the class.
+              * `host`: Host and optionally port of the WBEM server containing
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+
+        self.validate_namespace(namespace)
+        if isinstance(ObjectName, CIMInstanceName):
+            self._validate_instancename_namespace(namespace, ObjectName)
+            ref_paths = self._get_reference_instnames(namespace, ObjectName,
+                                                      ResultClass,
+                                                      Role)
+            rtn_names = [r.copy() for r in ref_paths]
+
+            for iname in rtn_names:
+                if iname.host is None:
+                    iname.host = self.host
+
+            return rtn_names
+
+        ref_classnames = self._get_reference_classnames(namespace, ObjectName,
+                                                        ResultClass, Role)
+
+        ref_result = [CIMClassName(classname=cn, host=self.host,
+                                   namespace=namespace)
+                      for cn in ref_classnames]
+
+        return ref_result
+
+    ####################################################################
+    #
+    #   References server request
+    #
+    ####################################################################
+
+    def References(self, namespace, ObjectName, ResultClass=None, Role=None,
+                   IncludeQualifiers=None, IncludeClassOrigin=None,
+                   PropertyList=None):
+        # pylint: disable=line-too-long
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.References`.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing slash
+            characters are ignored.
+
+          ObjectName:
+            The object path of the source object:
+
+            * Instance-level use: The instance path of the
+              source instance, as a :class:`~pywbem.CIMInstanceName` object.
+              If this object does not specify a namespace, the default namespace
+              of the connection is used.
```

Comment:
On "the default namespace of the connection": Change to "the namespace specified in the namespace parameter".

Reply:
DONE